### DISCUSSION TITLE
Using jotai atom in the QueryClient global cache-level callbacks

### DISCUSSION THREAD

**Discussion by nunoplima** [pmndrs/jotai#1941] [General]
Hi everyone!

In a Next.JS 12 app, I’d like to use the global cache-level callbacks when setting up the `QueryClient` and eventually fire a toast from the `onError` callback. The toasts are managed using Jotai. Having the Jotai Provider in the same file as the `QueryClientProvider`  does not seem to work, is there a way to achieve this? Having Jotai Provider in `_document.tsx` or the `QueryClientProvider` in a child component works, but I think I should  avoid those.
In the following codesandbox, the `<Home />` is trying to get the todos using the `useTodos` query hook (with a meta field setup) which fires an error, and ideally will be digested in `QueryClient` .
https://codesandbox.io/p/sandbox/unruffled-stonebraker-40iq89?file=%2Fpages%2F_app.tsx%3A40%2C14

Thank you for your time and attention.

**Comment by dai-shi** [comment]
Is it still an issue? Maybe @kalijonn can help?
